keras_retinanet/utils/eval.py

```diff
@@ -135,28 +135,20 @@ def _get_detections(generator, model, score_threshold=0.05, max_detections=100,
     # Returns
         A list of lists containing the detections for each image in the generator.
     """
-    all_detections = [[None for i in range(generator.num_classes()) if generator.has_label(i)] for j in range(generator.size())]
-
-    if geom_types is None:
-        geom_types = []
 
-    dir_names = [os.path.join(save_path, '{}s'.format(geom_type)) for geom_type in geom_types]
-    map(lambda name: os.makedirs(name) if not os.path.exists(name) else None, dir_names)
-
-    for i in progressbar.progressbar(range(generator.size()), prefix='Running network: '):
-        raw_image    = generator.load_image(i)
-        image        = generator.preprocess_image(raw_image.copy())
+    def _batch_preprocessing(generator, ind):
```

Comment:
Try to avoid **generator** transfer. It already exists in this scope.

```suggestion
    def _batch_preprocessing(ind):
```